pallets/pp/src/lib.rs

```diff
@@ -99,19 +182,35 @@ where
         caller: &T::AccountId,
         gas_limit: Weight,
         code: &Vec<u8>,
+        _utxo_hash: H256,
+        _utxo_value: u128,
         data: &Vec<u8>,
     ) -> Result<(), &'static str> {
         let code = pallet_contracts_primitives::Code::Upload(Bytes(code.to_vec()));
         let endowment = pallet_contracts::Pallet::<T>::subsistence_threshold();
 
-        let _ = pallet_contracts::Pallet::<T>::bare_instantiate(
+        let res = pallet_contracts::Pallet::<T>::bare_instantiate(
             caller.clone(),
             endowment * 100_u32.into(), // TODO
```

Comment:
The 100_u32 that is... Where does it come from?

Reply:
Each smart contract needs to have at least `subsistence_threshold` amount of balance so they do not get reaped but when you're creating the contract is has to be more than that, I don't know why. 100 is just something I put there to make the contract instantiate. I might as well be 1, 5 or 284 AFAIK. So no special meaning but this subsistence threshold is something we might be interested in having a discussion about at some point because it's very much tied to the balance code of Substrate, i.e., has not link to the UTXO stuff.